synfig-studio/src/gui/cellrenderer/cellrenderer_value.cpp

```diff
@@ -402,10 +402,17 @@ CellRenderer_ValueBase::render_vfunc(
 			Distance x( vector[0], Distance::SYSTEM_UNITS ), y( vector[1], Distance::SYSTEM_UNITS );
 			x.convert( App::distance_system, get_canvas()->rend_desc() );
 			y.convert( App::distance_system, get_canvas()->rend_desc() );
-			property_text() = strprintf("%s,%s", x.get_string(real_num_decimals).c_str(), y.get_string(real_num_decimals).c_str());
+			std::string text = strprintf("%s,%s", x.get_string(real_num_decimals).c_str(),
+												  y.get_string(real_num_decimals).c_str());
+			property_text() = text;
 		} else {
 			std::string format = strprintf("%%.%01df,%%.%01df", real_num_decimals, real_num_decimals);
-			property_text() = strprintf(format.c_str(), vector[0], vector[1]);
+			std::string text = strprintf(format.c_str(), vector[0], vector[1]);
```

Comment:
1) Avoid using keywords such as `final` as variable names.
2) This part can also be simplified. 
3) After completing step 2, check the changes again. You should notice a common part in all of them. I suggest to extract this part to the new function `std::string float_presentation(Real value, int num_decimals = real_num_decimals)`. This will allow you to move all formatting in one place.

Reply:
What about 3 step?

Reply:
> I was thinking about how to best implement it but Im still not sure how best to do it.

Step 2 can be done in a different way, this should lead you to a better solution.

Reply:
@ice0, how about now ? 

p.s. the formatting for the vector was a bit diff than the real it was `%%.%01d` vs `%%.%df` . I'm not sure what the 01 does but from comparison it seems like it added an extra decimal place so I think its probably not needed anyways, if not though we cold add a format argument to `float_presentation` what do you think ?

Reply:
Awesome!
This is exactly what I mean.

Next step is to check other places with `%%.%df` expression. For example:
```
if (type == type_angle)
{
	const std::string angle_format = strprintf("%%.%df°", angle_num_decimals);
	property_text() = strprintf( angle_format.c_str(), (Real) Angle::deg( data.get(Angle()) ).get() );
```

```
if (type == type_angle)
{
	property_text() = float_presentation(Angle::deg( data.get(Angle()) ).get(), angle_num_decimals) + "°";
```

>  I'm not sure what the 01 does

Same for me. 
@rodolforg Any ideas here?

Reply:
perfect, ok sounds good will do

Reply:
done, it seems like it was only there type_angle and transformation but for type_transformation it has two strings before and after so I don't think it would be any good to split it up to three parts.

Reply:
My suggestion:
```
property_text() = x.get_string(real_num_decimals) + "," + y.get_string(real_num_decimals) +
		float_presentation(angle.get(), angle_num_decimals) + "°" +
		sx.get_string(real_num_decimals) + "," + sy.get_string(real_num_decimals);
```